edgelet/api/identityServiceVersion_2020_06_01.yaml

```diff
@@ -0,0 +1,102 @@
+swagger: '2.0'
+schemes:
+  - http
+info:
+  title: Identity Service API
+  version: '2020-02-02'
+
+paths:
+  '/identities/identity':
+    get:
+      tags:
+        - Identity operations
+      summary: 'Get primary cloud identity for authenticated workload'
+      operationId: GetCurrentIdentity
+      parameters:
+        - $ref: '#/parameters/api-version'
+      responses:
+        '200':
+          description: Ok
+          schema:
+            $ref: '#/definitions/IdentityResult'
+        default:
+          description: Error
+          schema:
+            $ref: '#/definitions/ErrorResponse'
+
+definitions:
+  IdentityResult:
+    type: object
+    properties:
+      type:
+        type: string
+        description: The identity type.
+        enum:
+        - aziot
+      managedBy:
+        type: string
+        description: The owner of the identity.
+        enum:
+        - identityservice
+      spec:
+        $ref: '#/definitions/IdentitySpec'
+    required:
+      - type
+  IdentitySpec:
+    type: object
+    description: The identity specification.
+    properties:
+      hubName:
+        type: string
+        description: The name of the IoT hub where the device is provisioned.
+        example: "myhub.net"
+      deviceId:
+        type: string
+        description: The identity name of the provisioned device in the assigned IoT Hub.
+        example: "device01"
+      moduleId:
+        type: string
+        description: The identity name of the device workload in the provisioned device in the assigned IoT Hub.
+        example: "module01"
+      genId:
+        type: string
+        description: The generation ID of the device workload assigned by IoT Hub.
+        example: "12345"
+      auth:
+        $ref: '#/definitions/Credentials'
+    required:
+      - hubName
+      - deviceId
+      - managedBy
```

Comment:
Why this one is required? What can callers use the "managedBy" for?

Reply:
This is required in the future. So, we are including it now so callers don't need to change their code for this.

Reply:
Can you clarify what we use it for in the future?

Reply:
When module identities are generated on startup for host processes by identityservice, they will be having this field set as the current 'identityservice' enum. When module runtime requests for identities to be created for edge, they will be tagged as being managedby = 'edgeruntime' or 'moduleruntime'. This is used by 'moduleruntime' to delete identities it created, but it can still query for identities created by 'identityservice'.

Reply:
Ok. Given module runtime is the one driving the container module identity provisioning/lifetime; and IS for host processes, what's the benefit of having this field and mark it as required? Do I miss anything?

Reply:
So that we can setup policies for different CRUD operations on identities based on their owner. Managedby provides that context for each identity. It is just metadata and every system that automates identity management should have a clear owner for identity lifecycle. Module runtime being an external entity to identity service needs to have authorization and context to perform CRUD operations on identities created by identity service. We don't wish to hardcode these in code and couple the two services.

Reply:
For policy, shouldn't it be applied based on identity itself? Let's add it when it is clearer especially not having it as "required". Thanks.

Reply:
Removed this for now. It can be revisited later I agree.